src/commands/init.ts

```diff
@@ -72,13 +58,19 @@ export default class Init extends Command {
 
     ensureDirectoryExistence(flags.projectDir, true);
 
-    await this.initExpressProject();
+    const initializationType = await this.determineInitializationType(flags);
+    await this.initProject(flags, initializationType);
 
     const options = await this.getOptions();
 
     try {
       cli.action.start('Reading templates');
-      const files = readTemplates([path.resolve(__dirname, '..', 'templates', 'init')], flags.projectDir);
+      const excludes = initializationType === InitializationType.existingProject ? ['test'] : [];
+      const files = readTemplates({
+        from: [path.resolve(__dirname, '..', 'templates', 'init')],
+        to: flags.projectDir,
+        exclude: excludes
```

Comment:
Very minor thing but I would shorten `initializationType` to `initType` (same for the uppercase one) and move the `initType === InitType.existingProject ? ['test'] : [];` to line 72 so

``` ts
exclude: initType === InitType.existingProject ? ['test'] : []
```